core/src/main/java/org/neo4j/ogm/autoindex/AutoIndexManager.java

```diff
@@ -263,11 +265,16 @@ private static Set<AutoIndex> initialiseAutoIndex(MetaData metaData) {
 
             if (needsToBeIndexed(classInfo)) {
 
+                AnnotationInfo relationshipEntityAnnotation = classInfo.annotationsInfo().get(RelationshipEntity.class);
+                boolean isRelationShip = relationshipEntityAnnotation != null;
+
                 // We build the composite index first, to find out whether an @Id or @Index annotated field
                 // is actually decomposed by a MapCompositeConverter AND has a defined composite index.
                 Set<String> decomposedFields = new HashSet<>();
                 for (CompositeIndex index : classInfo.getCompositeIndexes()) {
-                    IndexType type = index.unique() ? IndexType.NODE_KEY_CONSTRAINT : IndexType.COMPOSITE_INDEX;
+                    IndexType type = index.unique() ?
```

Comment:
Could you please bring this into a `if`/`else` pattern? The nested ternary operation is hard to read.